File: iosApp/iosApp/AreaViewModel.swift
import Foundation
import SwiftUI
import Combine


class ViewModel: ObservableObject {
    static func areaCalculation(width: Double, height: Double) -> Double { width*height }
    
    static let initialWidth = 2.0
    static let initialHeight = 3.0
    
    @Published public var width = initialWidth
    @Published public var height = initialHeight
    @Published public var area: Double?
    
    init() {
        Publishers.CombineLatest($width, $height)
            .map(ViewModel.areaCalculation)
            .assign(to: &$area)
    }
}

File: androidApp/src/main/java/com/orange/areacalculator/androidApp/AreaViewModel.kt
package com.orange.areacalculator.androidApp

import androidx.lifecycle.ViewModel
import kotlinx.coroutines.flow.*


class AreaViewModel: ViewModel() {
    val areaCalculation = { width: Float, height: Float -> width*height }

    val initialWidth = 2.0f
    val initialHeight = 3.0f

    val widthFlow = MutableStateFlow(initialWidth)
    val heightFlow = MutableStateFlow(initialHeight)

    val setWidth: (Float) -> Unit = { value -> widthFlow.value = value }
    val setHeight: (Float) -> Unit = { value -> heightFlow.value = value }

    val areaFlow = combine(widthFlow, heightFlow, transform = areaCalculation)
}